lib/src/drawConfigContext.dart

```diff
@@ -0,0 +1,187 @@
+/// Copyright (c) 2017, the Dart Reddit API Wrapper  project authors.
+/// Please see the AUTHORS file for details. All rights reserved.
+/// Use of this source code is governed by a BSD-style license that
+/// can be found in the LICENSE file.
+
+import 'dart:io';
+import 'package:ini/ini.dart';
+import 'package:path/path.dart' as path;
+
+import './exceptions.dart';
+
+const String kFileName = 'praw.ini';
+const String kMacEnvVar = 'HOME';
+const String kLinuxEnvVar = 'XDG_CONFIG_HOME';
+const String kWindowsEnvVar = 'APPDATA';
+
+final kNotSet = null;
+
+///The [DrawConfigContext] class provides an iterface to store and load
+///the DRAW's configuration file [praw.ini].
+class DrawConfigContext {
+  ///Path to Local, User, Global Config Files, with matching precedence
+  Uri _localConfigPath;
+  Uri _userConfigPath;
+  Uri _globalConfigPath;
+
+  Config _customConfig;
+
+  Map<String, String> custom;
+
+  String _shortURL;
+  String _primarySiteName;
+
+  //Required fields for basic configuration
+  bool checkForUpdates;
+  String userAgent;
+
+  //Fields for Oauth workflow and configuration
+  String _redirectUri;
+  String clientId;
```

Comment:
These fields shouldn't be modifiable by the end user. Either these should all be final and populated in the constructor, or you need to do something like this:

`String get clientId => _clientId;`
`String _clientId;`